crates/generator/src/account_lock_manage/eip712/types.rs

```diff
@@ -64,9 +71,128 @@ impl EIP712Encode for WithdrawalAsset {
     }
 }
 
-// RawWithdrawalRequest
+#[derive(Debug)]
+pub enum AddressRegistry {
+    ETH,
+}
+
+impl AddressRegistry {
+    fn to_string(&self) -> &str {
+        "ETH"
+    }
+
+    pub fn from_registry_id(registry_id: u32) -> Result<Self> {
+        match registry_id {
+            ETH_REGISTRY_ACCOUNT_ID => Ok(Self::ETH),
+            _ => {
+                bail!("Unsupported registry id : {}", registry_id)
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct RegistryAddress {
+    registry: AddressRegistry,
+    address: [u8; 20],
+}
+
+impl RegistryAddress {
+    fn from_address(address: gw_common::registry_address::RegistryAddress) -> Result<Self> {
+        let registry = AddressRegistry::from_registry_id(address.registry_id)?;
+        if address.address.len() != 20 {
+            bail!(
+                "Invalid ETH address len, expected 20, got {}",
+                address.address.len()
+            );
+        }
+        Ok(RegistryAddress {
+            registry,
+            address: address.address.try_into().expect("eth address"),
+        })
+    }
+}
+
+impl EIP712Encode for RegistryAddress {
+    fn type_name() -> String {
+        "RegistryAddress".to_string()
+    }
+
+    fn encode_type(&self, buf: &mut Vec<u8>) {
+        buf.extend(b"RegistryAddress(string registry,address address)");
```

Comment:
string registry? uint?

Reply:
We use string in signature, so users can understand the signing content.